Lib/smtpd.py

```diff
@@ -86,6 +86,13 @@
     "SMTPChannel", "SMTPServer", "DebuggingServer", "PureProxy",
 ]
 
+warn(
+    'The smtpd module is deprecated and unmaintained.  Please see aiosmtpd '
+    '(https://aiosmtpd.readthedocs.io/) for the recommended replacement.',
+    DeprecationWarning,
+    stacklevel=2)
```

Comment:
This should probably be before the import asyncore. At the moment if you import smtpd you get a deprecation warning for asyncore, which is confusing (particularly with -Derror, where it exits without proceeding to give the error for smtpd as well).

Reply:
Good point, thanks.